src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -646,6 +646,17 @@ private void InitializeLuceneConfig()
             }
         }
 
+        /// <summary>
+        /// Apply the locale based on the user preference
+        /// </summary>
+        public static void ApplyLocaleAndNotify()
+        {
+            var preferenceFilePath = new PathManager().PreferenceFilePath;
```

Comment:
Why are we loading the preferences file again here? Can't we just read it from the Preferences model?

Reply:
> Why are we loading the preferences file again here, just to read which language is set? Can't we just read it from the Preferences model?

I don't think so because the Preferences are loaded here https://github.com/DynamoDS/Dynamo/blob/bbca4871cd140fe847dabc3be77599da7b7e4506/src/DynamoCore/Models/DynamoModel.cs#L683 as part of the StartDynamoWithDefaultConfig https://github.com/DynamoDS/Dynamo/blob/bbca4871cd140fe847dabc3be77599da7b7e4506/src/DynamoApplications/StartupUtils.cs#L279C35-L279C35, it's after the PreloadASM. We need the language before of that to notify to the user https://github.com/DynamoDS/Dynamo/blob/bbca4871cd140fe847dabc3be77599da7b7e4506/src/DynamoApplications/StartupUtils.cs#L277

Reply:
> Also does it default to english, when no language is set in preferences?

That's true because I'm reusing the Load function https://github.com/DynamoDS/Dynamo/blob/bbca4871cd140fe847dabc3be77599da7b7e4506/src/DynamoCore/Configuration/PreferenceSettings.cs#L976

Reply:
So in principle, we should load preferences and path manager a single time when launching Dynamo.

Reply:
@QilongTang @reddyashish Please your review, I have taken out the pathManager and preferences outside the DynamoModel so we can instantiate them before and reuse them as part of its ctor, any concern just let me know.